subsystem/elevator.py

```diff
@@ -0,0 +1,72 @@
+import rev
+
+from toolkit import Subsystem
+from toolkit.motors.rev_motors import SparkMax, SparkMaxConfig
+
+import config
+
+# TODO: CHANGE WHEN ROBOT IS BUILT
+ELEVATOR_CONFIG = SparkMaxConfig(
+    0.055, 0.0, 0.01, 0.000, (-.5, .75), idle_mode=rev.CANSparkMax.IdleMode.kBrake
+)
+
+class Elevator(Subsystem):
+
+    motor_extend: SparkMax = SparkMax(
+        config.elevator_can_id, config=ELEVATOR_CONFIG, inverted=False
+    )
+
+    def __init__(self) -> None:
+        super().__init__()
+        # Absolute encoder
+        self.zeroed: bool = False
+
+    def init(self) -> None:
+        self.motor_extend.init()
+        self.encoder: rev.SparkAbsoluteEncoder = rev.SparkAbsoluteEncoder(self.motor_extend.motor) # TODO: Make sure params work
+        self.motor_extend.motor.setClosedLoopRampRate(config.elevator_ramp_rate)
+
+    def set_length(self, length: float) -> None:
+        """
+        Sets the length of the elevator in meters
+
+        Args:
+            length (meters): Length in meters
+        """
+        
+        current_length = self.get_length()
+
+        if length > .3 * config.elevator_max_rotation:
+            ff = .65 * (1 if current_length < length else -1)
+        elif length < .3 * config.elevator_max_rotation and length < current_length:
+            ff = -.75
+        else: ff = 0
```

Comment:
Since this is an elevator, you will probably have constant FF that is the same going up and down. In that case, you would want to change the statement to if the new length is less than the old length (IE: elevator going down), and take the FF then

Reply:
Feels like this comment didn't finish????

Reply:
Sorry, I'll rewrite down here:

feedforward is a constant supply of voltage that the motor receives to offset a static force (gravity, static friction) from the PID loop. That way, the PID loop isn't always fighting the force in one direction, and working with the force in another. (in the case of gravity)

Since the elevator is always going against the force of gravity, you would have a constant feedforward that would only change from positive to negative when moving in different directions. 

If you were moving the elevator up, you would add a feedforward voltage on top of that PID, so when you stop the elevator, PID doesn't work to keep it from falling

on the flip side, if you were moving the elevator down, you would subtract the feedforward voltage on top of the PID, because the motor is moving in a different direction. 

for now, you can leave a random number in place in the constants, we will find out that number experimentally when we get our hands on it

Reply:
I am not sure of the correctness of this. I think that we should use the same feed-forward all the time. No matter which direction... but I could be wrong. I don't see why want to add power to the elevator going down when gravity is going to be doing some of the work for us.